sabnzbd/downloader.py

```diff
@@ -756,96 +776,110 @@ def run(self):
             if not read:
                 continue
 
-            for selected in read:
-                nw = self.read_fds[selected]
-                article = nw.article
-                server = nw.server
+            if self.recv_threads > 1:
+                for nw, bytes_received, done in self.recv_pool.imap_unordered(self.__recv, read):
+                    self.__handle_recv_result(nw, bytes_received, done)
+                if self.bandwidth_limit:
+                    self.__check_speed()
```

Comment:
I think we don't need 2 paths, with and without the pool. To 'disable' it can just be set to have 1 pool.
This way we also don't need the `__handle_recv_result`  and `__check_speed` functions, as they can just be inlined.

Reply:
Because of the overhead and also because I think it may be better for the bandwidth limiter if can sleep after each recv it checks instead of letting the threadpool clear all the buffers and then do a lot of sleeping.